### PR TITLE
chore(source-gcs): check AirbyteTracedException raised unit tests for file-based sources

### PR DESCRIPTION
**Pull request by strosek**
## What
This PR adds unit tests for `AirbyteTracedExceptions` raised during check_connection(). Tests will pass after this [PR](https://github.com/airbytehq/airbyte/pull/44438) is merged and a CDK is released.

## How
Add unit tests under `source-gcs` that checks for `AirbyteTracedException`s raised during check when one or multiple stream errors are found.

## Review guide
1. `test_source.py` - implements tests for single stream failure, and multiple stream failures during check.
2. `conftest.py` - test utilities

## User Impact
None

## Can this PR be safely reverted and rolled back?
- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-gcs/unit_tests/test_source.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+def _config_path():
+    return Path(__file__).resolve().parent.joinpath("resource/config/")
+
+
+def _mock_encoding_error():
+    DefaultFileBasedAvailabilityStrategy.check_availability_and_parsability = Mock(side_effect=AirbyteTracedException(message="encoding error"))
+
+
+def test_check_connection_with_bad_encoding(logger):
```
Comment (girarda) [inline airbyte-integrations/connectors/source-gcs/unit_tests/test_source.py:36]: from the source level, the error handling isn't specific to encoding, so let's make the test name generic

**Review by girarda — COMMENTED** [review]
small nit on the test name since the handling is generic. Can you also add a unit test for [the csv parser to verify a traced exception is now thrown](https://github.com/airbytehq/airbyte/pull/44438/files#diff-2ac011610fd803eed6b2f7f1369e9c9ecc6d99b01ebfb475781eec212d081064R59)

**Comment by strosek** [comment]
> small nit on the test name since the handling is generic. Can you also add a unit test for [the csv parser to verify a traced exception is now thrown](https://github.com/airbytehq/airbyte/pull/44438/files#diff-2ac011610fd803eed6b2f7f1369e9c9ecc6d99b01ebfb475781eec212d081064R59)

The test you mention was introduced in the original PR, [here](https://github.com/airbytehq/airbyte/pull/44438/files#r1735145896).

**Review by girarda — APPROVED** [review]
:shipit: !

**Pull request merged**